index.js

```diff
@@ -84,10 +85,37 @@ Lexer.prototype = {
    * @api private
    */
 
-  tok: function(type, val){
-    return val === undefined ?
-      {type: type, line: this.lineno} :
-      {type: type, line: this.lineno, val: val};
+  tok: function(type, val, columnIncrement){
+    var res = {type: type, line: this.lineno, col: this.colno };
+
+    if (val !== undefined) res.val = val;
+
+    this.incrementColumn(isNaN(columnIncrement) ? 0 : columnIncrement)
```

Comment:
@ForbesLindesay would you prefer me to refactor this line too so that column increment isn't passed into the function?